extensions/auth/portability-auth-google/src/main/java/org/dataportabilityproject/auth/google/GoogleAuthDataGenerator.java

```diff
@@ -44,10 +50,38 @@
  * authorization code and posts it against the AD API to obtain a token for querying the Graph API.
  */
 public class GoogleAuthDataGenerator implements AuthDataGenerator {
-  // TODO: Reduce requested scopes by service and authorization mode (readwrite/read)
-  private static final ImmutableCollection<String> SCOPES =
-      ImmutableSet.of(CalendarScopes.CALENDAR, PeopleServiceScopes.CONTACTS, TasksScopes.TASKS);
+  // The scopes necessary to import or export each supported data type.
+  // Scopes for EXPORT should contain READONLY permissions
+  private static final Map<String, ListMultimap<AuthMode, String>> DATA_TYPE_SCOPES =
+      ImmutableMap.<String, ListMultimap<AuthMode, String>>builder()
+          .put(
+              "calendar",
+              ImmutableListMultimap.<AuthMode, String>builder()
+                  .putAll(AuthMode.IMPORT, Arrays.asList(CalendarScopes.CALENDAR))
```

Comment:
I think if you call .put() you won't need the Arrays.asList()

Reply:
Done. I used ImmutableList.of() instead.